cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -817,20 +818,19 @@ pub enum Inst {
         kind: CondBrKind,
     },
 
-    /// A one-way conditional branch, invisible to the CFG processing; used *only* as part of
-    /// straight-line sequences in code to be emitted.
+    /// A conditional skip of an embedded Inst. This is a simple way to embed
+    /// control-flow (usually for trap-checks) in lowered sequences without
+    /// actually altering the CFG; the embedded branch is invisible at the CFG
+    /// level, because this pseudo-instruction encapsulates the whole single-in,
+    /// single-out region.
     ///
-    /// In more detail:
-    /// - This branch is lowered to a branch at the machine-code level, but does not end a basic
-    ///   block, and does not create edges in the CFG seen by regalloc.
-    /// - Thus, it is *only* valid to use as part of a single-in, single-out sequence that is
-    ///   lowered from a single CLIF instruction. For example, certain arithmetic operations may
-    ///   use these branches to handle certain conditions, such as overflows, traps, etc.
+    /// The `skip` `CondBrKind` gives the conditional-branch condition that will
+    /// *skip* the embedded `Inst`.
     ///
     /// See, e.g., the lowering of `trapif` (conditional trap) for an example.
-    OneWayCondBr {
-        target: BranchTarget,
+    CondSkip {
         kind: CondBrKind,
+        inst: Box<Inst>,
```

Comment:
As suggested by @julian-seward1 on zulip, do you think it'd make sense to keep it simple and stupid, and only assume a trap instruction as the embedded instruction? Only tests use a Nop4 instead of a Udf here, as far as I can tell, and it seems it would avoid a few headaches (what if the embedded instruction is a synthetic instruction, what if we compose several CondSkip within each other, potential early clobber problem buried inside this instruction...), and the use of a Box here.

If so, Julian's suggestion of `trapif` makes sense as a name, since it matches the CLIF IR instruction name, and real machine encodings on some other platforms.

Reply:
Yes, I like the idea of keeping this better-scoped -- changed to `Inst::TrapIf`. Thanks!